### PR TITLE
Fix incoming message size issue that introduced in #9113

### PR DESCRIPTION
**Pull request by codelipenghui**
### Motivation

Fix incoming message size issue that introduced in #9113. We should decrease the incoming message size when taking messages from the queue and increase the incoming message size while adding messages to the queue. With #9113, will always increase the incoming queue size.

### Modifications

Add method `increaseIncomingSize` and `decreaseIncomingSize`

### Verifying this change

Add a new test for verifying the incoming message size should be zero while the incoming queue size is zero.

### Does this pull request potentially affect one of the following parts:

*If `yes` was chosen, please highlight the changes*

  - Dependencies (does it add or upgrade a dependency): (no)
  - The public API: (no)
  - The schema: (no)
  - The default values of configurations: (no)
  - The wire protocol: (no)
  - The rest endpoints: (no)
  - The admin cli options: (no)
  - Anything that affects deployment: (no)

### Documentation

  - Does this pull request introduce a new feature? (no)

### CODE REVIEW

**Comment by codelipenghui** [comment]
@BewareMyPower @lhotari Please help review this PR

**Comment by BewareMyPower** [comment]
OK, I'll take a look soon.

File: pulsar-broker/src/test/java/org/apache/pulsar/client/api/SimpleProducerConsumerTest.java [hunk 1]
```diff
@@ -3692,4 +3693,88 @@ public void testGetStatsForPartitionedTopic() throws Exception {
...
+        });
+    }
+
+    @Test
+    public void testIncomingMessageSizeForPartitionedTopic() throws Exception {
+        final String topicName = "persistent://my-property/my-ns/testIncomingMessageSizeForPartitionedTopic-" +
+                UUID.randomUUID().toString();
+        final String subName = "my-sub";
+
+        admin.topics().createPartitionedTopic(topicName, 3);
```
Comment (BewareMyPower) [inline pulsar-broker/src/test/java/org/apache/pulsar/client/api/SimpleProducerConsumerTest.java:3744]: It's the only difference between `testIncomingMessageSizeForPartitionedTopic` and `testIncomingMessageSizeForNonPartitionedTopic`, IMO it's better to use `DataProvider` to reduce repeated logic, see `NullValueTest` for example.

**Review by BewareMyPower — CHANGES_REQUESTED** [review]

File: pulsar-broker/src/test/java/org/apache/pulsar/client/api/SimpleProducerConsumerTest.java [hunk 2]
```diff
@@ -3692,4 +3693,88 @@ public void testGetStatsForPartitionedTopic() throws Exception {
...
+        Producer<byte[]> producer = pulsarClient.newProducer()
+                .topic(topicName)
+                .create();
+
+        final int messages = 100;
+        List<CompletableFuture<MessageId>> messageIds = new ArrayList<>(messages);
+        for (int i = 0; i < messages; i++) {
+            messageIds.add(producer.newMessage().value(("Message-" + i).getBytes()).sendAsync());
+        }
+        FutureUtil.waitForAll(messageIds).get();
```
Comment (lhotari) [inline pulsar-broker/src/test/java/org/apache/pulsar/client/api/SimpleProducerConsumerTest.java:3719]: nitpick: in general, it's good to use timeouts in tests, for example using `.get(3, TimeUnit.SECONDS)`. However, I guess it's not very likely that sending would fail at this point, so the timeout isn't that relevant in this case.

**Review by lhotari — APPROVED** [review]

File: pulsar-client/src/main/java/org/apache/pulsar/client/impl/ConsumerBase.java [hunk 3]
```diff
@@ -847,13 +847,23 @@ protected boolean hasPendingBatchReceive() {
...
     }
 
-    protected void updateIncomingMessageSize(final Message<?> message) {
+    protected void decreaseIncomingMessageSize(final Message<?> message) {
         INCOMING_MESSAGES_SIZE_UPDATER.addAndGet(this,
-                (message.getData() != null) ? message.getData().length : 0);
+                (message.getData() != null) ? -message.getData().length : 0);
+    }
+
+    @VisibleForTesting
```
Comment (315157973) [inline pulsar-client/src/main/java/org/apache/pulsar/client/impl/ConsumerBase.java:864]: Public methods should not need this annotation

File: pulsar-broker/src/test/java/org/apache/pulsar/client/api/SimpleProducerConsumerTest.java (hunk 2 above)
Comment (sijie) [reply pulsar-broker/src/test/java/org/apache/pulsar/client/api/SimpleProducerConsumerTest.java:3719]: I don't think using timeout is a good idea. We should rely on the test framework timeout not the timeouts within the test.

Comment (lhotari) [reply pulsar-broker/src/test/java/org/apache/pulsar/client/api/SimpleProducerConsumerTest.java:3719]: @sijie yes, I agree that placing timeouts all over could clutter the test code. 
Do you happen to know if there's already a default timeout set for each individual test method?

**Comment by codelipenghui** [comment]
@BewareMyPower @315157973 I have addressed your comments, please take a look again.

**Review by BewareMyPower — APPROVED** [review]

**Review by lhotari — APPROVED** [review]

**Comment by 315157973** [comment]
The import of VisibleForTesting has not been deleted, I don’t know if the style check will report an error

File: pulsar-broker/src/test/java/org/apache/pulsar/client/api/SimpleProducerConsumerTest.java (hunk 2 above)
Comment (sijie) [reply pulsar-broker/src/test/java/org/apache/pulsar/client/api/SimpleProducerConsumerTest.java]: Yes. The TestNG has global timeout settings.

**Review by sijie — APPROVED** [review]

**Review by eolivelli — APPROVED** [review]
LGTM
